File: app/src/main/groovy/us/kesslern/ascient/service/PermissionHandlerService.java
package us.kesslern.ascient.service;

import android.Manifest;
import android.app.Activity;
import android.content.pm.PackageManager;
import android.support.v4.app.ActivityCompat;

/**
 * Request and manage permissions.
 */
public class PermissionHandlerService {
    private static final String RECEIVE_SMS = Manifest.permission.RECEIVE_SMS;
    private static final int PERMISSION_GRANTED = PackageManager.PERMISSION_GRANTED;

    public static void requestPermissions(Activity activity) {
        ActivityCompat.requestPermissions(activity, new String[]{Manifest.permission.RECEIVE_SMS}, 12);
        ActivityCompat.requestPermissions(activity, new String[]{Manifest.permission.INTERNET}, 12);
    }
}

File: app/src/main/java/us/kesslern/ascient/permission/PermissionHandlerService.kt
package us.kesslern.ascient.permission

import android.Manifest
import android.app.Activity
import android.content.pm.PackageManager
import android.support.v4.app.ActivityCompat

/**
 * Request and manage permissions.
 */
object PermissionHandlerService {

    fun requestPermissions(activity: Activity) {
        ActivityCompat.requestPermissions(activity, arrayOf(Manifest.permission.RECEIVE_SMS), 12)
        ActivityCompat.requestPermissions(activity, arrayOf(Manifest.permission.INTERNET), 12)
    }
}
